### ISSUE TITLE
Translate Discord Markdown formatting to IRC formatting

### ISSUE THREAD

**Issue by digitalcircuit** [sm00th/bitlbee-discord#28]
Most IRC clients support mIRC [color codes](http://en.wikichip.org/wiki/irc/colors ) and [formatting](https://wiki.mibbit.com/index.php/Formatting ) for *italics*, **bold**, etc.  For example, here's how [Quassel IRC handles converting from mIRC to Qt internal](https://github.com/quassel/quassel/blob/9585227400dd89f8a4d6bbc723f2d36ca0dc1a70/src/uisupport/uistyle.cpp#L580-L652 ).

Discord also [supports formatting (minus colors) via their flavor of Markdown](https://support.discordapp.com/hc/en-us/articles/210298617-Markdown-Text-101-Chat-Formatting-Bold-Italic-Underline- ).  Mapping mIRC to Markdown and vice-versa would help keep visual intent the same.

This should be optional, akin to the ```incoming_me_translation```, say, ```incoming_format_translation```, and apply to both channels and private messages (*also see issue #27*).  Sending out mIRC codes probably should always be translated as the special characters are nonsensical to the Discord client.

Some more information on color and formatting (feel free to ignore the color aspect):
http://en.wikichip.org/wiki/irc/colors
https://wiki.mibbit.com/index.php/Formatting
http://www.mirc.com/colors.html

*Aside: the Bitlbee OTR plugin strips certain color combinations and the prefix of* __```<<OTR>>```__ *in order to ensure OTR messages cannot be spoofed.  However, it's unlikely to need handled here, as Discord itself doesn't support colors.*

**Comment by zertap** [comment]
Yes, translating incoming and outgoing bolds and italics would be a nice addition.

**Comment by dgw** [comment]
I don't know how long this has been the case, since I only started using bitlbee-discord last week, but I get `<@root> discord - Error: Failed to send message (400).` in `&bitlbee` when trying to send messages that contain (m)IRC formatting characters.

Outbound translation seems wise. Inbound is less important, because most of us can probably interpret the formatting characters sent by Discord through our experiences with Markdown in various applications.

**Comment by digitalcircuit** [comment]
> Outbound translation seems wise. Inbound is less important, because most of us can probably interpret the formatting characters sent by Discord through our experiences with Markdown in various applications.

I agree on the relative importance, and if it's too much hassle to make incoming translation optional, feel free to leave it out.  It'd just seem odd to do [IRC format parsing](http://modern.ircdocs.horse/formatting.html ) at all then only parse it one way, instead of **optionally** both ways.
